feat: add Pipocas.tv subtitle provider to Bazarr

Blackspirits opened the pull request:
- Implement PipocasProvider with login, search and download support
- Add language mapping (pt, pt-BR, en, es)
- Auto-register provider via subliminal_patch loader
- Add Pipocas.tv configuration entry to UI (username/password)
- Include archive extraction and scoring logic

Blackspirits committed:
feat: add Pipocas.tv subtitle provider to Bazarr

- Implement PipocasProvider with login, search and download support
- Add language mapping (pt, pt-BR, en, es)
- Auto-register provider via subliminal_patch loader
- Add Pipocas.tv configuration entry to UI (username/password)
- Include archive extraction and scoring logic

morpheus65535 commented:
Can you add this provider to the README.md file also please?

worm69 commented:
Hi guys, i am testing the changes but i find some issues with login
<img width="1297" height="878" alt="image" src="https://github.com/user-attachments/assets/46ecbb09-8757-4cb0-aaa2-fc1a494271e7" />

I will try check better the issue, if i mock the user&pass on code is working

Pipocas.tv :: searching '91 Days S01E12' [portugues]

diogormendes commented:
@Blackspirits ?

Blackspirits committed:
Fix Pipocas provider integration

Blackspirits commented:
Updated.

Changes made:
- Added Pipocas.tv to README
- Added missing `pipocas.username` and `pipocas.password` settings validators
- Added Pipocas credentials to `get_providers_auth()`
- Updated archive handling to use `ProviderSubtitleArchiveMixin`

The login issue should now be fixed because credentials are passed from Bazarr settings to the provider instead of needing to be hardcoded.

morpheus65535 commented on custom_libs/subliminal_patch/providers/pipocas.py:
```diff
@@ -0,0 +1,397 @@
+            archive_stream.seek(0)
+
+            if is_zipfile(archive_stream):
+                archive_stream.seek(0)
```
Why?

morpheus65535 commented on custom_libs/subliminal_patch/providers/pipocas.py:
```diff
@@ -0,0 +1,397 @@
+            archive_stream.seek(0)
+            archive = RarFile(archive_stream)
+        else:
+            archive_stream.seek(0)
```
Why?

morpheus65535 commented on custom_libs/subliminal_patch/providers/pipocas.py:
```diff
@@ -0,0 +1,397 @@
+        archive_stream = io.BytesIO(data)
+
+        if is_rarfile(archive_stream):
+            archive_stream.seek(0)
```
Why do you seek the archive?

morpheus65535 commented on custom_libs/subliminal_patch/providers/pipocas.py:
```diff
@@ -0,0 +1,397 @@
+                subtitle.content = fix_line_ending(data)
+
+                if subtitle.is_valid():
+                    subtitle.normalize()
```
Why normalize yourself when subliminal will do it later?

morpheus65535 commented on custom_libs/subliminal_patch/providers/pipocas.py:
```diff
@@ -0,0 +1,397 @@
+        if not subtitle.content:
+            raise ServiceUnavailable("Pipocas.tv :: no suitable subtitle file in archive")
+
+        subtitle.normalize()
```
Why?

Blackspirits committed:
Address Pipocas provider review comments

Blackspirits commented:
Updated.

Removed the unnecessary archive stream `seek(0)` calls and removed manual `subtitle.normalize()` calls from the provider, since normalization is handled later by subliminal.

morpheus65535 commented:
<img width="776" height="238" alt="image" src="https://github.com/user-attachments/assets/46f4c320-0607-4960-a4c9-b34feff8b787" />

Blackspirits committed:
Format provider list

morpheus65535 commented:
Thanks!

worm69 commented:
thank you both! @Blackspirits @morpheus65535 i will do some new testing
